app/forms/utilisation_report.rb

```diff
@@ -11,31 +11,39 @@ def initialize(params = {})
   end
 
   def generate
-    bookable, blocked = bookable_and_blocked
     CSV.generate do |csv|
       csv << [:booked_appointments, :bookable_slots, :blocked_slots]
-      csv << [Appointment.where(start_at: range).count, bookable, blocked]
+      range.each do |day|
+        csv << generate_for_day(day)
+      end
     end
   end
 
   def file_name
-    integer_range = range ? "#{range.begin.to_i}-#{range.end.to_i}" : nil
-    "utilisation-report-#{integer_range}.csv"
+    "utilisation-report-#{range_title}.csv"
   end
 
   private
 
-  def bookable_and_blocked
+  def generate_for_day(day)
+    day_range = (day.beginning_of_day..day.end_of_day)
+    bookable, blocked = bookable_and_blocked(day_range)
+    [
+      day,
+      Appointment.where(start_at: day_range).count,
+      bookable,
+      blocked
+    ]
+  end
+
+  def bookable_and_blocked(day_range)
+    slots_within_range = slots_within_range(day_range)
```

Comment:
Maybe call this `slots` or something as it's a bit weird in that it shadows the method at first glance.